### PR TITLE
Processing managed in poolmanager

### PR DESCRIPTION
**Pull request by gjn**
I little re-factoring.

* New Poolmanager class
* Remove multiprocessing option in config file (we don't use it anymore)
* Ability to create stats without tapping into db (might be slow for high zoom levels) with `make statsnodb`

### CODE REVIEW

**Comment by gjn** [comment]
I've re-created level 18 with this PR:

`It took 9:04:24.439480 to create 737342 tiles (55379 were skipped)`

Last run for level 18 mentioned in https://github.com/geoadmin/3d-forge/pull/105 was:
`It took 14:26:17.328468 to create 737392 tiles (55342 were skipped)`

Faster by 37%. But might also be because of lookup fix by Loic and other optimisations.

**Comment by gjn** [comment]
@loicgasser This PR is ready to review/merge.

File: forge/lib/poolmanager.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    def __init__(self):
+        self._pool = multiprocessing.Pool(NUMBER_POOL_PROCESSES, initProcess)
+
+    def _abort(self):
+        self._pool.terminate()
+        self._pool.join()
+
+    def numOfProcesses(self):
+        return NUMBER_POOL_PROCESSES
```
Comment (loicgasser) [inline forge/lib/poolmanager.py]: numOfProcesses, could be intitialized in the __init__ and could take per default the number of CPU or a custom nb of pool process. Minor though

File: forge/lib/tiler.py [hunk 2]
```diff
@@ -226,59 +192,34 @@ def create(self):
...
-                try:
-                    worker(j)
-                except Exception as e:
-                    logger.error('An error occured while generating the tile: %s' % e)
-                    raise Exception(e)
+        tiles = Tiles(self.tmsConfig, self.t0)
+
+        pm = PoolManager()
+
+        useChunks = self.chunks
```
Comment (loicgasser) [inline forge/lib/tiler.py]: I don't see why you need a new variable here, self.chunks should reflect the actual size of the chunks being used.

File: tms.cfg [hunk 3]
```diff
@@ -1,11 +1,8 @@
 [General]
-# multiProcessing: 0 -> no multiprocessing
-# multiProcessing: 1 -> multiprocessing based on the nb of processors
-multiProcessing: 1
-# number of chunks to pass to processing pool
-chunks: 2500
 # terrain files base path
 bucketpath: 1.0.0/ch.swisstopo.terrain.3d/default/20151231/4326/
+# chunks per process (that's a maximum)
+chunks: 50
```
Comment (loicgasser) [inline tms.cfg]: let's rename it maxChunkSize,  no?

Comment (loicgasser) [inline tms.cfg]: or maxChunks

**Comment by loicgasser** [comment]
Nice simplification, some minor comments. Thanks a lot

**Pull request closed without merging**